move/interchain_token_service/sources/versioned/interchain_token_service_v0.move

```diff
@@ -200,6 +211,58 @@ module interchain_token_service::interchain_token_service_v0 {
         (token_id, treasury_cap_reclaimer)
     }
 
+    public(package) fun link_coin(
+        self: &InterchainTokenService_v0,
+        deployer: &Channel,
+        salt: Bytes32,
+        destination_chain: String,
+        destination_token_address: vector<u8>,
+        token_manager_type: u8,
+        link_params: vector<u8>,
+    ): MessageTicket {
+        assert!(destination_token_address.length() != 0, EEmptyTokenAddress);
+
+        // Custom token managers can't be deployed with native interchain token type, which is reserved for interchain tokens
+        assert!(token_manager_type != token_manager_types::native_interchain_token(), ECannotDeployInterchainTokenManager);
+
+        // Cannot deploy to this chain using linkToken
+        assert!(destination_chain.as_bytes().length() != 0, ENotSupported);
+
+        let chain_name_hash = self.chain_name_hash;
+
+        // Cannot deploy to this chain using linkToken remotely
+        assert!(chain_name_hash != bytes32::from_bytes(keccak256(destination_chain.as_bytes())), ECannotDeployRemotelyToSelf);
```

Comment:
don't think this is needed, the hub should prevent this, and this should fail on execution on receiving part in Sui, anyways? EVM had this for legacy token linking reasons